homeworks/homework-1/app/src/features/CourseListPage.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import {
+	ListGroup,
+	Button,
+	Form,
+	FormControl,
+	FormGroup,
+} from 'react-bootstrap';
+import {CourseListItem} from "../components/CourseListItem";
+import {Header} from "../components/common/Header";
+import  './CourseListPage.css';
+
+class CourseListPage extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      courses: [
+        {
+          id: 1,
+          title: 'CourseForm 1',
+          duration: '180',
+          date: '12.12.12',
+          description: 'Description1'
+        },
+        {
+          id: 2,
+          title: 'CourseForm 2',
+          duration: '20',
+          date: '12.12.12',
+          description: 'Description2'
+        },
+        {
+          id: 3,
+          title: 'CourseForm 3',
+          duration: '100',
+          date: '12.12.12',
+          description: 'Description3'
+        }
+      ],
+      title: "Courses"
+    };
+  }
+
+  render() {
+    const {
+      courses
+    } = this.state;
+
+    return (
+      <div>
+        <Header
+					pageTitle={this.state.title}
+				/>
+
+        <Form inline>
+          <FormGroup controlId="searchParameter">
+            <FormControl
+              type="text"
+              placeholder="enter name or date"/>
+          </FormGroup>
+          <Button type="submit">Search</Button>
+          <Button type="submit">Add course</Button>
+        </Form>
+
+        <ListGroup componentClass="ul">
+          {this.renderCourses(courses)}
```

Comment:
do not pass `courses` as argument, use `this.state.courses` exactly in function

Reply:
pass just that arguments that not accessible inside component, for example if you render courses list like this `{this.state.courses,map((course, index) => this.renderCourse(course, index))}` in this case it needed to pass each course to function.
Also that better to move outside template:
```
render() {
  const corsesList = this.state.courses,map((course, index) => this.renderCourse(course, index));
  
  return (
  <div>
    .....
    {corsesList }
  </div>
  );
}
```